### PR TITLE
Block Page 2.0

### PR DESCRIPTION
**Pull request by WaLLy3K**
**By submitting this pull request, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

10

---
The two primary goals of re-writing the block page is to keep the styling consistent with the Admin Console, and to minimise the use of Javascript as we should not expect users to have JS enabled on known-bad domains.

The changes within `index.php` are dependant on an alteration of the output of `queryFunc()` within `/usr/local/bin/pihole`, as well as `/var/www/html/admin/scripts/pi-hole/php/queryads.php`

### CODE REVIEW

**Comment by dschaper** [comment]
`index.js` is no longer needed?

**Comment by WaLLy3K** [comment]
Just to clarify, #1420 removes the Javascript redirect from `lighttpd.conf`, and [this line](https://github.com/pi-hole/pi-hole/pull/1416/commits/551add5f447b25e7c5c3e54c325b287212dfd9c8#diff-622eb992a761acc873605fcdfbaf1e3bR79) will redirect users to a blank but valid Javascript filler.

**Comment by WaLLy3K** [comment]
Have already  [implemented the fix](https://github.com/pi-hole/pi-hole/pull/1416/files#diff-622eb992a761acc873605fcdfbaf1e3bR300) as proposed in #1452.

**Comment by WaLLy3K** [comment]
Remaining commit order:
* #1456 Update queryFunc() to search Whitelist
* [AdminLTE #488](https://github.com/pi-hole/AdminLTE/pull/488) Supporting aLTE change for core tweak/blockpage branch 
* This PR, #1416
* #1420 Disable HTTPS & shift functions to block page

**Comment by WaLLy3K** [comment]
Note to self: confirm and fix issue #1479 before continuing

**Comment by WaLLy3K** [comment]
Renamed [WIP], awaiting merge of [AdminLTE #488](https://github.com/pi-hole/AdminLTE/pull/488).

File: advanced/index.php [hunk 1]
```diff
@@ -1,224 +1,335 @@
...
+    if (isset($type) && $type === "js") header("Content-Type: application/javascript");
 }
 
-if(in_array($uriExt, $webExt) || empty($uriExt))
-{
-	// Requested resource has an extension listed in $webExt
-	// or no extension (index access to some folder incl. the root dir)
-	$showPage = true;
+// Determine block page redirect
+if ($serverName === "pi.hole") {
```
Comment (diginc) [inline advanced/index.php]: Shouldn't all of the AUTHORIZED_HOSTNAMES be redirected?
e.g. localhost, 127.0.0.1, ::1, and $VIRTUAL_HOST variable contents (for my docker ;))

Comment (WaLLy3K) [inline advanced/index.php]: So the logic is that we need to determine if the website that's being visited is one for the Pi, or one on a block list.

Virtual Host will redirect to the splash page (Local Pi landing page) by [L42](https://github.com/WaLLy3K/pi-hole/blob/69fa9874dd2a05e83225220fd46aa0a425f32aa3/advanced/index.php#L42), while everything else is covered by checking if it's a valid IP in [L68](https://github.com/WaLLy3K/pi-hole/blob/69fa9874dd2a05e83225220fd46aa0a425f32aa3/advanced/index.php#L68).

File: advanced/index.js [hunk 2]
```diff
@@ -1 +1 @@
-var x = "Pi-hole: A black hole for Internet advertisements."
+var x = ""
```
Comment (DL6ER) [inline advanced/index.js]: You can remove the file in another PR (where the source branch is on the team repo). I can adjust the tests for your.

**Comment by WaLLy3K** [comment]
Currently required merges to support this PR:
* [Supporting aLTE change for core tweak/blockpage branch](https://github.com/pi-hole/AdminLTE/pull/488)
* [Disable HTTPS & shift functions to block page](https://github.com/pi-hole/pi-hole/pull/1420)

All are ready to go.

**Comment by PromoFaux** [comment]
We still need to get around the installer not updating the block page on people's installs.

I guess we could just make the decision to not support custom block pages, in order to ensure that the block page the user is running is always up to date.

Failing that, could we add a file detection/redirect to this block page? e.g if `/var/www/html/pihole/customBlock.php` exists, redirect to that instead of displaying this one...

**Comment by Mcat12** [comment]
I like the idea of having a `customBlock.php` file override the default instead of having to do some strange change detection.

**Comment by PromoFaux** [comment]
Approved

**Pull request merged**